### PR TITLE
chore(ci): support pkg.pr.new

### LINKED ISSUE #7754: [Feature]: Support for pkg.pr.new
### What problem does this feature solve?

Add support for https://github.com/stackblitz-labs/pkg.pr.new

This enables users to access features and bug-fixes without the need to wait for release cycles using npm or pull request merges.

### What does the proposed API of configuration look like?

Add GitHub actions.

### PR DESCRIPTION
**Pull request by shulaoda**
## Summary

closes #7754

## Checklist

- [ ] Tests updated (or not required).
- [ ] Documentation updated (or not required).

### CODE REVIEW

**Comment by shulaoda** [comment]
Could you check [https://github.com/apps/pkg-pr-new](https://github.com/apps/pkg-pr-new)? @chenjiahan 
```bash
Run npx pkg-pr-new publish --compact --pnpm './npm/*' './crates/node_binding' './packages/rspack'
{"url":"/check","statusCode":404,"statusMessage":"Not Found","message":"The app https://github.com/apps/pkg-pr-new is not installed on web-infra-dev/rspack."}
Error: Process completed with exit code 1.
```

**Comment by chenjiahan** [comment]
@shulaoda I just reinstalled the app, can you try again?

image

File: .github/workflows/preview-commit.yml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        run: pnpm run build:js
+
+      - name: Move artifacts
+        run: node scripts/build-npm.cjs
+
+      - name: Show binding packages
+        run: ls -R npm
+
+      - name: Release
+        run: API_URL=https://pkg.pr.new pnpm dlx pkg-pr-new@https://7b32ad52.stackblitz-cr.pages.dev/stackblitz-labs/pkg.pr.new/playground-b@234 publish --compact --pnpm './npm/*' './crates/node_binding' './packages/rspack'
```
Comment (Aslemammad) [inline .github/workflows/preview-commit.yml:72]: maybe you should put API_URL in `env:`, but if it runs then no need

File: package.json [hunk 2]
```diff
@@ -72,6 +73,7 @@
     "jest-cli": "29.7.0",
     "jest-environment-node": "29.7.0",
     "lint-staged": "^15.0.0",
+    "pkg-pr-new": "workspace:*",
```
Comment (Aslemammad) [inline package.json:76]: I guess you'd have to remove this too!

File: .github/workflows/preview-commit.yml (hunk 1 above)
Comment (shulaoda) [reply .github/workflows/preview-commit.yml:72]: Okay, but there are currently some issues with the assembly line. I will try again later.

**Comment by shulaoda** [comment]
Can you try this?

Installing https://github.com/apps/continuous-releases-staging

And then running:
```
pnpm dlx pkg-pr-new@https://05ece7b9.stackblitz-cr.pages.dev/stackblitz-labs/pkg.pr.new/pkg-pr-new@227 publish --compact --pnpm './npm/*' './crates/node_binding' './packages/rspack'
```

But make sure to remove that app after trying so you don't have the staging github app.

_Originally posted by @Aslemammad in https://github.com/stackblitz-labs/pkg.pr.new/issues/223#issuecomment-2343943198_
            
@chenjiahan cc

**Comment by SoonIter** [comment]
is it ready?

**Comment by shulaoda** [comment]
> is it ready?

Not yet, can you help me install https://github.com/apps/continuous-releases-staging? I need to test.

**Comment by chenjiahan** [comment]
done 😄

image

**Comment by shulaoda** [comment]
> is it ready?

ready now

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -391,6 +391,12 @@ jobs:
...
     name: Pkg Preview
-    # needs:
-    #   - cargo-deny
+    needs:
+      - test-linux
+      - test-windows
+      - cargo-deny
+      - lint
+      - rust_check
+      - rust_test
```
Comment (SoonIter) [inline .github/workflows/ci.yml:400]: Is this automatically triggered in all mrs?

`rust_test` needs `rust_changes`, so `pkg-preview` is skipped ?

File: .github/workflows/preview-commit.yml [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+        run: pnpm run build:js
+
+      - name: Move artifacts
+        run: node scripts/build-npm.cjs
+
+      - name: Show binding packages
+        run: ls -R npm
+
+      - name: Release
+        run: npx pkg-pr-new publish --compact --pnpm './npm/*' './crates/node_binding' './packages/rspack'
```
Comment (SoonIter) [inline .github/workflows/preview-commit.yml:72]: skip `@rspack/cli`?

**Review by SoonIter — APPROVED** [review]
LGTM

**Comment by chenjiahan** [comment]
Awesome 👍🏻

**Pull request merged**